### PR TITLE
feat(background-agent): add profile parameter to spawn

### LINKED ISSUE #100: Add configurable profiles to background agents
## Summary

Extend `background_agent spawn` with an optional `profile` parameter that selects a different system prompt + model combination, instead of always inheriting the spawning persona's config.

This enables building specialized agent teams — code reviewers, test writers, architects — without new communication infrastructure.

## Context

Technical proposal: `cf-notes/2026-03-27-inter-persona-communication-proposal.md` (Option 1)

## What Changes

- **`background-agent.ts`**: Add optional `profile` parameter to spawn action
- **`background-agent-manager.ts`**: Load profile config when spawning instead of inheriting parent persona
- **`talond.yaml` or `profiles/` directory**: Profile definitions (name, model, system prompt file, skills, capabilities)
- **`persona-loader.ts`**: Minor extension to load profiles (reuse persona loading logic)

## Example Profiles

- `senior-engineer` — code review, architecture feedback, PR analysis
- `test-writer` — integration/unit test generation
- `code-reviewer` — focused code review with security/performance lens
- `architect` — high-level design review, RFC feedback

## Example Usage

```
background_agent spawn profile="senior-engineer" task="Review PR #42, focus on error handling edge cases"
background_agent spawn profile="test-writer" task="Write integration tests for the new tool"
```

## Effort Estimate

1-2 days. Most infrastructure already exists.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by ivo-toby**
## Summary
- Adds optional `profile` parameter to `background_agent spawn`
- When set, spawned agent uses the named persona's system prompt + model + skills instead of the thread's persona
- Unknown profiles return a descriptive error listing available options
- Model is passed through to the provider only when no explicit provider override is given (prevents cross-provider model mismatch)
- No profile = existing behavior (backward compat)

Closes #100

## What changed
- `src/tools/host-tools/background-agent.ts`: added `profile?` to spawn args; profile lookup + error handling; model passthrough
- `src/tools/host-tools-mcp-server.ts`: added `profile` and `provider` to MCP tool schema
- `src/subagents/background/background-agent-manager.ts`: added `profileName?` and `model?` to `SpawnBackgroundAgentInput`
- `src/providers/provider-types.ts`: added `model?` to `ProviderSpawnInput`
- `src/providers/claude-code-provider.ts`: pass `--model` flag when model override is provided
- `src/personas/persona-loader.ts`: added `listNames()` method
- `config/talond.example.yaml`: documented profile usage
- `tests/unit/tools/background-agent.test.ts`: 4 new profile tests (valid profile, unknown profile, no profile, provider override suppresses model)

## Test plan

**Prerequisites:** `npm install`

1. Run unit tests: `npx vitest run tests/unit/tools/background-agent.test.ts`
   - Expected: 15 tests pass, including 4 new profile tests

2. Run full suite: `npm test`
   - Expected: all tests pass (1 pre-existing failure in `list-skills.test.ts` unrelated to this PR)

3. Manual (with a running daemon):
   - Configure two personas in talond.yaml (e.g. `assistant` and `code-reviewer`)
   - Spawn with profile: `background_agent spawn profile="code-reviewer" prompt="Summarise this repo"`
     - Expected: agent uses code-reviewer's system prompt, model, and provider
   - Spawn with unknown profile: `background_agent spawn profile="nonexistent" prompt="..."`
     - Expected: error message listing available profile names
   - Spawn without profile: `background_agent spawn prompt="Hello"`
     - Expected: uses thread's persona (unchanged behavior)
   - Spawn with profile + explicit provider override: `background_agent spawn profile="code-reviewer" provider="claude-code" prompt="..."`
     - Expected: uses claude-code provider but does NOT pass the profile's model (avoids cross-provider mismatch)

🤖 Generated with Claude Code

### CODE REVIEW

File: src/tools/host-tools/background-agent.ts [hunk 1]
```diff
@@ -169,6 +190,10 @@ export class BackgroundAgentHandler {
           : typeof loadedPersona.config.provider === 'string' && loadedPersona.config.provider.trim().length > 0
             ? loadedPersona.config.provider.trim()
             : undefined,
+      ...(args.profile ? { profileName: args.profile } : {}),
+      // Only pass the persona's model when the provider is NOT explicitly overridden.
+      // Cross-provider model names (e.g. "claude-opus-4-6" on gemini-cli) would be invalid.
+      ...(!args.provider && loadedPersona.config.model ? { model: loadedPersona.config.model } : {}),
```
[the remark this answers is not in the document]
Comment (ivo-toby) [reply src/tools/host-tools/background-agent.ts]: Fixed in a490bdb: updated `GeminiCliProvider.prepareBackgroundInvocation` to honour `input.model` via `--model` flag when set, matching claude-code-provider behaviour. Added test coverage.

File: src/cli/commands/add-persona.ts [hunk 2]
```diff
@@ -135,14 +163,19 @@ export async function addPersona(options: AddPersonaOptions): Promise<AddPersona
...
     systemPromptFile,
-    skills: [],
+    skills: options.skills ?? [],
     capabilities: {
-      allow: [],
-      requireApproval: [],
+      allow: options.capabilities ?? [],
+      requireApproval: options.requireApproval ?? [],
     },
   };
```
[the remark this answers is not in the document]
Comment (talond-dev) [reply src/cli/commands/add-persona.ts]: Already addressed in earlier commits (a891c74): model is omitted from YAML when `--provider` is set without `--model`. The config schema default is a separate issue tracked in #106.
